read-fonts/src/table_provider.rs

```diff
@@ -101,3 +101,56 @@ pub trait TableProvider<'a> {
             .and_then(FontRead::read)
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    /// https://github.com/googlefonts/fontations/issues/105
+    #[test]
+    fn bug_105() {
+        // serve some dummy versions of the tables used to compute hmtx. The only
+        // fields that matter are maxp::num_glyphs and hhea::number_of_h_metrics,
+        // everything else is zero'd out
+        struct DummyProvider;
+        impl TableProvider<'static> for DummyProvider {
+            fn data_for_tag(&self, tag: Tag) -> Option<FontData<'static>> {
+                if tag == Tag::new(b"maxp") {
```

Comment:
beginner question: would it not be more "idiomatic" to write this if/else block as a match statement?

Reply:
It would, but unfortunately `match` does not allow us to have a 'method call' in a pattern, even if it is a `const` method, so we can't match against `Tag::new(b"xxxx")`, and would need to have `if` guards anyway, like:

```rust
match tag {
    t if t == Tag::new(b"maxp") => (),
   // etc
}
```

which I wouldn't hate, but it doesn't feel like exactly a match statement? it is a bit more concise though..

Reply:
oh i see. thanks for the explanation!